src/macros.jl

```diff
@@ -1032,23 +1027,24 @@ macro variable(args...)
             var = x.args[2]
             var isa Number && _error("Variable declaration of the form `$(x.args[3]) $(x.args[1]) $var` is not supported. Use `$(x.args[3]) <= $var` instead.")
             @assert length(x.args) == 3
-            lb = esc_nonconstant(x.args[3])
-            haslb = true
-            ub = Inf
+            setlowerbound(infoexpr, esc_nonconstant(x.args[3]), _error)
+            if !infoexpr.hasub
+                infoexpr.upperbound = Inf # Change NaN -> Inf which has same type
```

Comment:
Is this still needed? (Same below)

Reply:
I don't think so, let's remove it

Reply:
You meant the comment or the whole three lines ?

Reply:
I removed the lines, the `upperbound` value does not matter since `hasub` is false.